File: backend/src/routes/health.ts
import { Router, Request, Response } from 'express';
import { prisma } from '../db/prisma';

const router = Router();

// ─── GET /api/health — Health check ─────────────────────
router.get('/', async (_req: Request, res: Response): Promise<void> => {
  try {
    // Check database connectivity
    await prisma.$queryRaw`SELECT 1`;

    res.json({
      success: true,
      data: {
        status: 'healthy',
        timestamp: new Date().toISOString(),
        uptime: process.uptime(),
      },
    });
  } catch (error) {
    res.status(503).json({
      success: false,
      error: 'Service unavailable — database connection failed',
    });
  }
});

export default router;

File: backend/app/routers/health.py
"""
health.py — Health check endpoint.
Replaces: backend/src/routes/health.ts
"""

from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse
from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db

router = APIRouter(prefix="/api/health")


@router.get("")
async def health_check(db: AsyncSession = Depends(get_db)):
    try:
        await db.execute(text("SELECT 1"))
        return {
            "success": True,
            "data": {
                "status": "healthy",
                "database": "connected",
            },
        }
    except Exception as e:
        return JSONResponse(
            status_code=503,
            content={
                "success": False,
                "error": "Database connection failed",
            },
        )
